demo/lib/orchestrator.go

```diff
@@ -496,29 +500,34 @@ func (e *Orchestrator) RunResharing(resharingGroup *ResharingGroup, timeout time
 		fmt.Printf("\t- Joining DKG for node %s\n", n.PrivateAddr())
 		err = n.JoinReshare(oldGroup)
 		if err != nil {
-			panic(err)
+			return nil, err
 		}
 		fmt.Printf("\t- Joined DKG for node %s\n", n.PrivateAddr())
 	}
 
+	done := make(map[string]bool)
 	for _, n := range except(e.reshareNodes[1:], e.newNodes) {
 		n := n
+		if n.Index() == leader.Index() || done[n.PublicAddr()] {
+			continue
+		}
```

Comment:
Can `done[n.PublicAddr()]` be true at any time in that loop? 😮 
Are there duplicated nodes in `except(e.reshareNodes[1:], e.newNodes)`?

Reply:
for some reason this was failing - I didn't think there could be duplicate nodes, but this solved it